### PR TITLE
Add database set/query and cap query

### PR DESCRIPTION
**Pull request by yuvalavn**

### CODE REVIEW

File: drivers/common/mlx5/mlx5_prm.h [hunk 1]
```diff
@@ -2081,6 +2081,106 @@ struct mlx5_ifc_query_virtq_out_bits {
...
+};
+
+struct mlx5_ifc_set_regexp_params_in_bits {
+	u8         opcode[0x10];
+	u8         uid[0x10];
+
+	u8         reserved_at_20[0x10];
+	u8         op_mod[0x10];
+
+	u8         reserved_at_40[0x18];
```
Comment (orikam) [inline drivers/common/mlx5/mlx5_prm.h:2111]: please remove empty line.

File: drivers/common/mlx5/mlx5_prm.h [hunk 2]
```diff
@@ -2081,6 +2081,106 @@ struct mlx5_ifc_query_virtq_out_bits {
 	struct mlx5_ifc_virtio_net_q_bits virtq;
 };
 
+
+struct regexp_params_field_select_bits {
+	u8         db_umem_id[0x1];
+	u8         stop_engine[0x1];
+	u8         reserved_at_2[0x1e];
+};
+
```
Comment (orikam) [inline drivers/common/mlx5/mlx5_prm.h:2095]: small comment relevant for all of this patch. 
coding convention in DPDK there is no tab/space between the type and the field name.

File: drivers/regex/meson.build [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2017 Intel Corporation
```
Comment (orikam) [inline drivers/regex/meson.build:2]: You don't work at Intel :) please fix for all files

File: drivers/regex/Makefile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2010-2015 Intel Corporation
+
```
Comment (orikam) [inline drivers/regex/Makefile:3]: You don't work at Intel

File: drivers/regex/mlx5/mlx5_regex.c [hunk 5]
```diff
@@ -3,3 +3,15 @@
  */
 
 #include "mlx5_regex.h"
+
+int mlx5_regex_is_supported(struct ibv_context *ibv_ctx)
+{
+	struct mlx5_hca_attr caps;
+	int err;
+
+	err = mlx5_devx_cmd_query_hca_attr(ibv_ctx, &caps);
```
Comment (orikam) [inline drivers/regex/mlx5/mlx5_regex.c:13]: The caps should be saved and reused, There is no point of reading the caps for each bit.
I assume that there will be other caps that we will need.

**Review by orikam — CHANGES_REQUESTED** [review]

File: drivers/common/mlx5/mlx5_prm.h (hunk 2 above)
Comment (yuvalavn) [reply drivers/common/mlx5/mlx5_prm.h:2095]: Do we have style checker script?

**Pull request closed without merging**